### PR TITLE
Mfcc model, training, and plot

### PR DESCRIPTION
**Pull request by jack-jintao-huang**
- Added mfcc model
- added training function
- added plot function
- fixed matplotlib memory leak issue

### CODE REVIEW

File: src/cnn/mfcc_model.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        # Calculate the size of the input tensor to the fully connected layer
+        self.fc_input_size = self.calculate_fc_input_size()
+
+        self.fc1 = nn.Linear(self.fc_input_size, 512)
+        self.relu5 = nn.ReLU()
+        self.fc2 = nn.Linear(512, 10)
+
+    def calculate_fc_input_size(self):
+        # Perform a forward pass to get the output shape of the last convolutional layer
```
Comment (tmahabir) [inline src/cnn/mfcc_model.py:33]: Wouldn't it be more efficient/better practice to use the output formulas we've learned in class to determine the output size of the last convolutional layer

Comment (jack-jintao-huang) [reply src/cnn/mfcc_model.py:33]: yes it would be. This formula makes my life easier when I was messing around with the convolution settings / layers. And in terms of efficiency this is only called once during initialization, so it isn't too big of a performance hit. Does it help with your concern?

Comment (tmahabir) [reply src/cnn/mfcc_model.py:33]: Yes, good point about it only being called once👍🏽

**Review by tmahabir — APPROVED** [review]
lgtm

File: src/utils/plot.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    plt.show()
+    plt.title("Train vs Validation Loss")
+    plt.plot(range(1,n+1), train_loss, label="Train")
+    plt.plot(range(1,n+1), val_loss, label="Validation")
+    plt.xlabel("Epoch")
+    plt.ylabel("Loss")
+    plt.legend(loc='best')
+    plt.show()
+
+def plot_confusion_matrix(model_path, classes):
```
Comment (nicholasshort) [inline src/utils/plot.py:32]: How does this ./csv path work? Is this specific to your local machine or does it work with colab?

I also think it might be useful to have this function create the confusion matrix as well. We could add as parameters the prediction and label arrays. Maybe that's for the future, but good to keep in mind.

Comment (jack-jintao-huang) [reply src/utils/plot.py:32]: the ./csv works for both colab and local machine. I've used it for my labs.

The function takes in `classes` that is used to label the prediction and label array. I will create a docstring in the future.

Comment (nicholasshort) [reply src/utils/plot.py:32]: Thanks!

**Review by nicholasshort — APPROVED** [review]
Looks Good!
